orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -825,6 +819,206 @@ void AddNewScalarNodeArgAndInitializer(Graph& graph,
   graph.AddInitializedTensor(proto_data);
 }
 
+Status FindAllConnectedNodes(Graph& graph,
+                             const Node* node,
+                             std::vector<const Node*>& connected_nodes,
```

Comment:
Is there a reason to use a vector in some cases here, and a set in others?

Reply:
If an initializer is consumed by a node twice, setting `connected_inputs` to `set` can avoid duplicates.

---
In reply to: [452254431](https://github.com/microsoft/onnxruntime/pull/4321#discussion_r452254431) [](ancestors = 452254431)